seqr/views/apis/project_categories_api_tests.py

```diff
@@ -4,15 +4,41 @@
 from seqr.views.apis.project_categories_api import update_project_categories_handler
 from seqr.views.utils.test_utils import _check_login
 
+PROJECT_GUID = 'R0001_1kg'
+PROJECT_CAT_GUID2 = 'PC000002_categry_with_unicde'
+PROJECT_CAT_GUID3 = 'PC000003_test_category_name'
+PROJECT_CAT_GUID4 = 'PC000004_new_project_category'
+PROJECT_CAT_GUID5 = 'PC000005_pc000002_categry_with'
+NEW_PROJECT_CAT_NAME = 'New project category'
+
 
 class ProjectCategoriesAPITest(TestCase):
     fixtures = ['users', '1kg_project']
 
     def test_project_categories_api(self):
-        url = reverse(update_project_categories_handler, args=['R0001_1kg'])
+        url = reverse(update_project_categories_handler, args=[PROJECT_GUID])
         _check_login(self, url)
 
         response = self.client.post(url, content_type='application/json', data=json.dumps({
             'categories': []
         }))
         self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+
+        self.assertListEqual(response_json.keys(), ['projectsByGuid', 'projectCategoriesByGuid'])
+        self.assertListEqual(response_json['projectsByGuid'].keys(), [PROJECT_GUID])
+        self.assertListEqual(response_json['projectCategoriesByGuid'].keys(), [PROJECT_CAT_GUID3, PROJECT_CAT_GUID2])
+        self.assertIsNone(response_json['projectCategoriesByGuid'][PROJECT_CAT_GUID2])
+        self.assertIsNone(response_json['projectCategoriesByGuid'][PROJECT_CAT_GUID3])
+
+        response = self.client.post(url, content_type='application/json', data=json.dumps({
+            'categories': ['PC000002_categry_with_unicde', NEW_PROJECT_CAT_NAME]
+        }))
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+
+        self.assertListEqual(response_json.keys(), ['projectsByGuid', 'projectCategoriesByGuid'])
+        self.assertListEqual(response_json['projectsByGuid'].keys(), [PROJECT_GUID])
+        self.assertListEqual(response_json['projectCategoriesByGuid'].keys(), [PROJECT_CAT_GUID4, PROJECT_CAT_GUID5])
```

Comment:
You should never assume you deterministically know what the guid that is generated will be. In the past, tests like that have proven to be incredibly flappy/ unreliable. Instead, you should assert that the expected known guid is in the response, and that there are 2 guids returned. Then you should take the other, new guid and query the database for it and check that the object exist and has the name you expect. And then you can check the project json and make sure it has the expected 2 guids

Reply:
Okay. I thought the results are always determined for a test setup.

Reply:
But the guid generation could change, we could add another test fixture and the ID would increment. In general, we want tests to be as stable as possible and since we don't care what the guid is we don't want to write fragile tests that rely on it always being the same

Reply:
I updated the test result examination with guids from the model.